opensrp-pnc/src/main/java/org/smartregister/pnc/config/PncCloseFormProcessing.java

```diff
@@ -1,40 +1,238 @@
 
-        JSONArray fieldsArray = PncUtils.generateFieldsFromJsonForm(jsonFormObject);
-        FormTag formTag = PncJsonFormUtils.formTag(PncUtils.getAllSharedPreferences());
+        try {
+            ArrayList<Event> eventList = new ArrayList<>();
+            JSONObject jsonFormObject = new JSONObject(jsonString);
+
+            JSONArray fieldsArray = PncUtils.generateFieldsFromJsonForm(jsonFormObject);
+            FormTag formTag = PncJsonFormUtils.formTag(PncUtils.getAllSharedPreferences());
+
+            JSONObject metadata = jsonFormObject.getJSONObject(METADATA);
+
+            String baseEntityId = PncUtils.getIntentValue(data, PncConstants.IntentKey.BASE_ENTITY_ID);
+            String entityTable = PncUtils.getIntentValue(data, PncConstants.IntentKey.ENTITY_TABLE);
+            Event closePncEvent = JsonFormUtils.createEvent(fieldsArray, metadata, formTag, baseEntityId, eventType, entityTable);
+            PncJsonFormUtils.tagSyncMetadata(closePncEvent);
+            eventList.add(closePncEvent);
+
+            if ("woman_died".equals(getFieldValue(fieldsArray, "pnc_close_reason"))) {
+                closePncEvent.setEventType(PncConstants.EventTypeConstants.DEATH);
+
+//                addSaveReportDeceasedObservations(fieldsArray, closePncEvent);
+//                updateMetadata(metadata, closePncEvent);
+//
+//                String encounterDateField = getFieldValue(fieldsArray, "date_of_death");
+//                Date encounterDate = new Date();
+//                String encounterDateTimeString = null;
+//                if (StringUtils.isNotBlank(encounterDateField)) {
+//                    encounterDateTimeString = formatDate(encounterDateField);
+//                    Date dateTime = formatDate(encounterDateField, false);
+//                    if (dateTime != null) {
+//                        encounterDate = dateTime;
+//                    }
+//                }
+
+                createDeathEventObject(PncLibrary.getInstance().context().applicationContext(), closePncEvent.getProviderId(), closePncEvent.getLocationId(), closePncEvent.getEntityType(), PncLibrary.getInstance().eventClientRepository(), closePncEvent, eventType, entityTable);
+
+//                ContentValues values = new ContentValues();
+//                values.put(PncConstants.KeyConstants.DOD, encounterDateField);
+//                values.put(PncConstants.KeyConstants.DATE_REMOVED, PncUtils.getTodaysDate());
+//                updateChildFTSTables(values, baseEntityId);
+//
+//                updateDateOfRemoval(baseEntityId, encounterDateTimeString);
+
+            }
+
+            return eventList;
+        }
+        catch (Exception ex) {
+            Timber.e(ex);
+            return new ArrayList<>();
+        }
+    }
+
+    private void createDeathEventObject(Context context, String providerId, String locationId, String entityId, EventClientRepository db, Event event, String encounterType, String entityTable) throws JSONException {
```

Comment:
you use the event argument passed here and call the property you need

Reply:
`createDeathEventObject(Context context, String providerId, String locationId, String entityId, EventClientRepository db, Event event, String encounterType, String entityTable)` ->  `createDeathEventObject(Event event)`

Reply:
done